core/src/main/java/com/scalar/db/transaction/autocommit/AutoCommitTransactionManager.java

```diff
@@ -0,0 +1,115 @@
+package com.scalar.db.transaction.autocommit;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.Isolation;
+import com.scalar.db.api.SerializableStrategy;
+import com.scalar.db.api.TransactionState;
+import com.scalar.db.common.ActiveTransactionManagedDistributedTransactionManager;
+import com.scalar.db.common.error.CoreError;
+import com.scalar.db.config.DatabaseConfig;
+import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.db.service.StorageFactory;
+import java.util.UUID;
+
+public class AutoCommitTransactionManager
+    extends ActiveTransactionManagedDistributedTransactionManager {
+
+  private final DistributedStorage storage;
+
+  public AutoCommitTransactionManager(DatabaseConfig databaseConfig) {
+    super(databaseConfig);
+    StorageFactory storageFactory = StorageFactory.create(databaseConfig.getProperties());
+    storage = storageFactory.getStorage();
+  }
+
+  @VisibleForTesting
+  AutoCommitTransactionManager(DatabaseConfig databaseConfig, DistributedStorage storage) {
+    super(databaseConfig);
+    this.storage = storage;
+  }
+
+  @Override
+  public DistributedTransaction begin() throws TransactionException {
```

Comment:
Just wondering what will happen when doing `begin()` with the auto-commit mode.
I miss something, but I don't think the generated transaction ID will be really tied to the actual transaction　since it is not stored in the coordinator table, so I don't really get the use case of this method.

Reply:
Right. The transaction ID is basically never used. So, it's a pseudo one actually.

Reply:
The current transaction abstraction is something as follows:

```java
DistributedTransactionManager transactionManager = ...;

DistributedTransaction transaction = transactionManager.begin();

// CRUD operations
...

transaction.commit();
```

So, we can't disable the `begin()` method actually in this abstraction. What do you think?

Reply:
Then, maybe outputting warning messages?
It would be great if we could chat!